app/views/samples/_form_assays_js.haml

```diff
@@ -3,19 +3,28 @@
     dropzoneOptions = {
       paramName: "assay_files",
       autoProcessQueue: true,
+      maxThumbnailFilesize: 20,
       parallelUploads: 10,
       url: '/assay_files/create',
       uploadMultiple: true,
+      renameFile: function (file) {
+        var completeFileName = file.name;
+        var sanitizedName = completeFileName.replace(/[^a-z^A-Z^0-9\^_.-]/g, '_');
+        var fileName = sanitizedName.split('.')[0];
+        return sanitizedName.replace(fileName, fileName + Date.now());
```

Comment:
There may be dots inside the filename (e.g. `file.name.pdf)`), so you'll want to split and skip the last entry:

```js
var fileName = sanitizedName.split(".").slice(0, -1).join(".")
```

Thought it's possible that it may not even be the extension, which means that you won't sanitize the end of the filename, which could be a problem. So if you need to inject a timestamp:

```js
var parts = sanitizedFilename.split(".")
var extname = parts.pop()
parts.push(Date.now())
parts.push(extname)
return filename
```

But again: storing the files as a mere UUID would be a better solution.